src/app/CommandHandler.cpp

```diff
@@ -113,6 +113,17 @@ Status CommandHandler::OnInvokeCommandRequest(CommandHandlerExchangeInterface &
     return status;
 }
 
+CHIP_ERROR CommandHandler::TryAddResponseData(const ConcreteCommandPath & path, CommandId commandId,
+                                              DataModel::EncoderToTLV & encoder)
+{
+    ConcreteCommandPath responseCommandPath = { path.mEndpointId, path.mClusterId, commandId };
+    ReturnErrorOnFailure(TryAddResponseDataPreEncode(path, responseCommandPath));
```

Comment:
The split between TryAddResponseData and TryAddResponseDataPreEncode was an artificial distinction due to the templating: TryAddResponseDataPreEncode was non-templated and shared across instantiations.

Do we still want/need this split?  Is there a reason to have TryAddResponseDataPreEncode be a separate function?  And does it need to be inlined?  The documentation for this function no longer makes sense in the new setup, by the way; the whioe "non-templated... " bit is not relevant.